girder/__init__.py

```diff
@@ -74,6 +76,28 @@ def format(self, record, *args, **kwargs):
         return super(LogFormatter, self).format(record, *args, **kwargs)
 
 
+class StreamToLogger:
+    """
+    Redirect a file-like stream to a logger.
+    """
+    def __init__(self, stream, logger, level):
+        self.stream = stream
+        self.logger = logger
+        self.level = level
+        self.logger._girderLogHandlerOutput = False
+        for key in dir(stream):
+            if key != 'write' and not key.startswith('_') and callable(getattr(stream, key)):
+                setattr(self, key, getattr(stream, key))
```

Comment:
We are overriding the default `sys.stdout` and `sys.stderr` with the logger method.  However, I want to preserve as much functionality for stdout and stderr as possible, so that other modules that send data to them can do so without a problem.  The only method we really need to override is `write`, but if we cannot mutate the `write` method on the stream itself, so we replace the stream with this custom class.  To preserve the stream methods, all of them get added to our class instance *except* private and built-in methods, which, in python, begin with `_`.

Fundamentally, it lets our stream replacement handle functions `flush`, `writeline`, and others without having to enumerate them individually.